src/server/test/routes/unitEditParamsTest.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+const { expect } = require('chai');
+const { chai, mocha, app, testDB } = require('../common');
+const Unit = require('../../models/Unit');
+const { insertUnits } = require('../../util/insertData');
+const { getUnitIdByName } = require('../../util/readingsUtils');
+const {
+	validateString,
+	validateInt,
+	validateBool,
+	validateMinMaxRelation
+} = require('../util/vaidationHelpers');
+
+const EDIT_UNIT = '/api/units/edit';
+
+const basePayload = {
+	name: 'Valid Name',
+	identifier: 'valid_id',
+	unitRepresent: 'flow',
+	secInRate: 60,
+	typeOfUnit: 'unit',
+	suffix: 'L/s',
+	displayable: 'all',
+	preferredDisplay: true,
+	note: 'Note text',
+	minVal: 0,
+	maxVal: 100,
+	disableChecks: 'reject_bad'
+};
+
+
+const INSERT_UNIT = {
+	name: 'kWh',
+	identifier: '',
+	unitRepresent: Unit.unitRepresentType.QUANTITY,
+	secInRate: 3600,
+	typeOfUnit: Unit.unitType.UNIT,
+	suffix: '',
+	displayable: Unit.displayableType.ALL,
+	preferredDisplay: true,
+	note: 'OED created standard unit'
+};
+
+
+mocha.describe('Unit Routes - /edit Validation', () => {
+
+	// SHL: I put a lot of comment in unitParamsTest.js which probably is now obsolete but some apply here.
+	mocha.beforeEach(async () => {
+		const conn = testDB.getConnection();
+
+		await insertUnits([INSERT_UNIT], true, conn);
+	});
+
+	mocha.it('should validate string fields', async () => {
+		await validateString({
+			field: 'name',
+			endpoint: EDIT_UNIT,
+			basePayload,
+			minLength: 1,
+			maxLength: 50
+		});
+
+		await validateString({
+			field: 'identifier',
+			endpoint: EDIT_UNIT,
+			basePayload,
+			minLength: 1,
+			maxLength: 50,
+			required: true
+		});
+
+		await validateString({
+			field: 'unitRepresent',
+			endpoint: EDIT_UNIT,
+			basePayload,
+			enumValues: Object.values(Unit.unitRepresentType),
+			minLength: 1
```

Comment:
The enums do not have a minLength in add and it should be consistent. If you check all valid and an invalid one (such as empty) maybe it isn't needed. I don't think you do that yet but per my comment on checking correct values that might be nice.